### PR TITLE
feat(jsonnet-bundler)!: Use absolute import format for depName

### LINKED ISSUE #13568: jsonnet-bundler: Use absolute import format as dependency names
### What would you like Renovate to be able to do?

The `jsonnet-bundler` manager should use the absolute import format for dependency names similar to Golang, this would be inline with the direction taken by the project ([jsonnet-bundler/jsonnet-bundler#61](https://github.com/jsonnet-bundler/jsonnet-bundler/issues/61))

The manager currently has an inconsistent use of the legacy import behavior which:

* Uses the legacy `name` field
* Fallbacks to base path of `source.git.remote / source.git.subdir` (**inconsistency**: the Renovate manager uses base path of `source.git.remote`)

This behavior has been called legacy since [jsonnet-bundler/jsonnet-bundler#63](https://github.com/jsonnet-bundler/jsonnet-bundler/pull/63)

Current implementation for quick reference:

https://github.com/renovatebot/renovate/blob/b60996672b3d99a1ebdfc2e2d24486d832095e74/lib/manager/jsonnet-bundler/extract.ts#L51-L52

For example, this package:

```json
       "source": {
         "git": {
           "remote": "https://github.com/grafana/jsonnet-libs.git",
           "subdir": "memcached-mixin"
         }
       }
```

Currently `depName=jsonnet-libs`, and the absolut import format would be `depName=github.com/grafana/jsonnet-libs/memcached-mixin`

Related discussion #13529

### If you have any ideas on how this should be implemented, please tell us here.

`source.git.remote` is an URL, so it could be parsed with the [`URL`](https://nodejs.org/api/url.html) module instead of using a regex, the resulting `depName` would be the following composition:

```typescript
import { posix } from 'path';
import { URL } from 'url';

const gitRemote = new URL(dependency.source.git.remote);

const depName = posix.join(
  gitRemote.host,
  gitRemote.pathname.replace(/\.git$/, ''),
  dependency.source.git.subdir,
)
```

Points my opinion is strong about:

* the manager should not use the legacy `name` field
* `depName` should include more than repository base name for uniqueness

Points I still have a few doubts, but leaning toward strong:

* `depName` should include `subdir`
  * On one hand, it is consistent with Go packages and `remote+subdir` is what describe a package in the `jsonnetfile.json`, and would guarantee uniqueness 
  * On the other hand, people may prefer macro-updates per repos, but using package rules may be better for that

### Is this a feature you are interested in implementing yourself?

Yes

### PR DESCRIPTION
**Pull request by maxbrunet**
## Changes

BREAKING CHANGE: `depName` now uses the "absolute import" format (e.g. `bar` → `github.com/foo/bar/baz-wow`)

## Context

Closes #13568

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by rarkins** [comment]
@maxbrunet can you resolve the git conflict?

**Comment by rarkins** [comment]
@maxbrunet sorry, can you resolve the conflicts?

**Review by rarkins — APPROVED** [review]

File: lib/modules/manager/jsonnet-bundler/extract.ts [hunk 1]
```diff
@@ -46,11 +44,16 @@ function extractDependency(dependency: Dependency): PackageDependency | null {
     return null;
   }
 
-  const match = gitUrl.exec(dependency.source.git.remote);
+  const gitRemote = new URL(dependency.source.git.remote);
```
Comment (viceice) [inline lib/modules/manager/jsonnet-bundler/extract.ts:47]: don't use deprecated `url` module, use `parseUrl` from utils

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/jsonnet-bundler/extract.ts (hunk 1 above)
Comment (maxbrunet) [reply lib/modules/manager/jsonnet-bundler/extract.ts:47]: Looks like @rarkins started working on the branch, I am not sure if it is ok for me to make changes until this is released, and there are unrelated changes now, maybe a follow up PR afterward?

Comment (rarkins) [reply lib/modules/manager/jsonnet-bundler/extract.ts:47]: I think it's just a side effect of us rebasing v33. I did "Update Branch" and it seems back to normal for you to continue

File: lib/modules/manager/jsonnet-bundler/extract.ts [hunk 2]
```diff
@@ -1,13 +1,11 @@
+import { URL } from 'url';
```
Comment (viceice) [inline lib/modules/manager/jsonnet-bundler/extract.ts:1]: we should at least remove this import and use the global URL object 
```suggestion
```

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/modules/manager/jsonnet-bundler/__fixtures__/jsonnetfile-empty-git-source.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (viceice) [inline lib/modules/manager/jsonnet-bundler/__fixtures__/jsonnetfile-empty-git-source.json:1]: can we please inline this small fixture ( io perf )

File: lib/modules/manager/jsonnet-bundler/extract.ts [hunk 4]
```diff
@@ -1,13 +1,11 @@
+import { join } from 'upath';
+
```
Comment (viceice) [inline lib/modules/manager/jsonnet-bundler/extract.ts:2]:
```suggestion
```

File: lib/modules/manager/jsonnet-bundler/extract.ts [hunk 5]
```diff
@@ -46,11 +44,19 @@ function extractDependency(dependency: Dependency): PackageDependency | null {
     return null;
   }
 
-  const match = gitUrl.exec(dependency.source.git.remote);
+  const gitRemote = parseUrl(dependency.source.git.remote);
+  if (gitRemote === null) {
+    return null;
```
Comment (viceice) [inline lib/modules/manager/jsonnet-bundler/extract.ts:49]: log a debug message when url is invalid for visibility

File: .devcontainer/Dockerfile [hunk 6]
```diff
@@ -1,4 +1,4 @@
-FROM containerbase/node:14.20.1@sha256:e275fb819c433373dbcdd7e7fc7c4fbdf6a08880e66eb9617ff311b6766e6a2c
+FROM containerbase/node:16.17.1
```
Comment (rarkins) [inline .devcontainer/Dockerfile:1]:
```suggestion
FROM containerbase/node:14.20.1@sha256:e275fb819c433373dbcdd7e7fc7c4fbdf6a08880e66eb9617ff311b6766e6a2c
```

**Review by rarkins — APPROVED** [review]

**Review by viceice — APPROVED** [review]

**Pull request merged**